### PR TITLE
Overhaul fakes to use a JSON file on disk

### PR DESCRIPTION
**Pull request by paultag**

### CODE REVIEW

**Comment by paultag** [comment]
The idea here is I'm going to write a script to take a VACOLS record and turn that into a JSON file we can use to test "prod" records without hitting prod

File: app/models/fakes.rb [hunk 1]
```diff
@@ -140,199 +140,34 @@ def upload_form8(first_name, middle_initial, last_name, file_name)
...
-        correspondent: Correspondent.new(
-          'Joe Snuffy', 'Self', 'Joe Snuffy', 'Joe', '', 'Snuffy',
-        ),
-        bfha: '1',
-      ),
-    }
+    def self.load_data
+      fakes = "app/fixtures/cases.json"
+      ENV.has_key?('CASEFLOW_FAKES') do
+          fakes = ENV['CASEFLOW_FAKES']
```
Comment (alex) [inline app/models/fakes.rb]: 2 space indent. also this syntax is incorrect, `has_key()` doesn't take a block AFAIK?

File: app/models/fakes.rb [hunk 2]
```diff
@@ -140,199 +140,34 @@ def upload_form8(first_name, middle_initial, last_name, file_name)
...
+      create_case = lambda do |data|
+        Caseflow::Fakes::Case::DATE_FIELDS.each do |field|
+          if data.has_key?(field)
+             data[field] = Date.parse(data[field])
+          end
+        end
+        data[:folder] = Folder.new(data[:folder])
+        data[:correspondent] = Correspondent.new(*data[:correspondent])
+        Case.new(**data)
+      end
```
Comment (alex) [inline app/models/fakes.rb]: Just factor this into a function, no need to make a lambda

File: app/models/fakes.rb (hunk 1 above)
Comment (paultag) [inline app/models/fakes.rb]: Arh, thanks

File: app/models/fakes.rb [hunk 3]
```diff
@@ -140,199 +140,39 @@ def upload_form8(first_name, middle_initial, last_name, file_name)
...
+
+      File.open(fakes, "r") do |f|
+        data = JSON.load(f).deep_symbolize_keys
+        data.map {|k, v| [k.to_s, Caseflow::Fakes::case_from_json(v)]}.to_h
+      end
+    end
+
+    def self.data
+      @DATA ||= load_data
+      @DATA
```
Comment (alex) [inline app/models/fakes.rb]: This line isn't needed, `@DATA ||=` line returns `@DATA`

Comment (paultag) [inline app/models/fakes.rb]: :+1:

**Pull request merged**